internal/providers/pluginfw/pluginfw_rollout_utils.go

```diff
@@ -0,0 +1,209 @@
+	if useSdkV2 == "" {
+		return []string{}
+	}
+	return strings.Split(useSdkV2, ",")
+}
+
+// GetUseSdkV2DataSources is a helper function to get name of data sources that should use SDK V2 instead of plugin framework
+func getUseSdkV2DataSources() []string {
+	useSdkV2 := os.Getenv("USE_SDK_V2_DATA_SOURCES")
+	if useSdkV2 == "" {
+		return []string{}
+	}
+	return strings.Split(useSdkV2, ",")
+}
+
+// Helper function to check if a resource should use be in SDK V2 instead of plugin framework
+func shouldUseSdkV2Resource(resourceName string) bool {
+	useSdkV2Resources := getUseSdkV2Resources()
+	for _, sdkV2Resource := range useSdkV2Resources {
+		if resourceName == sdkV2Resource {
+			return true
+		}
+	}
+	return false
+}
+
+// Helper function to check if a data source should use be in SDK V2 instead of plugin framework
+func shouldUseSdkV2DataSource(dataSourceName string) bool {
+	sdkV2DataSources := getUseSdkV2DataSources()
+	for _, sdkV2DataSource := range sdkV2DataSources {
+		if dataSourceName == sdkV2DataSource {
+			return true
+		}
+	}
+	return false
+}
+
+// getPluginFrameworkResourcesToRegister is a helper function to get the list of resources that are migrated away from sdkv2 to plugin framework
+func getPluginFrameworkResourcesToRegister(sdkV2Fallbacks ...SdkV2FallbackOption) []func() resource.Resource {
+	fallbackOption := sdkV2FallbackOptions{}
+	for _, o := range sdkV2Fallbacks {
+		o.Apply(&fallbackOption)
+	}
+
+	var resources []func() resource.Resource
+
+	// Loop through the map and add resources if they're not specifically marked to use the SDK V2
+	for _, resourceFunc := range migratedResources {
+		name := getResourceName(resourceFunc)
+		if !shouldUseSdkV2Resource(name) && !slices.Contains(fallbackOption.resourceFallbacks, name) {
+			resources = append(resources, resourceFunc)
+		}
+	}
+
+	return append(resources, onboardedResources...)
+}
+
+// getPluginFrameworkDataSourcesToRegister is a helper function to get the list of data sources that are migrated away from sdkv2 to plugin framework
+func getPluginFrameworkDataSourcesToRegister(sdkV2Fallbacks ...SdkV2FallbackOption) []func() datasource.DataSource {
+	fallbackOption := sdkV2FallbackOptions{}
+	for _, o := range sdkV2Fallbacks {
+		o.Apply(&fallbackOption)
+	}
+
+	var dataSources []func() datasource.DataSource
+
+	// Loop through the map and add data sources if they're not specifically marked to use the SDK V2
+	for _, dataSourceFunc := range migratedDataSources {
+		name := getDataSourceName(dataSourceFunc)
+		if !shouldUseSdkV2DataSource(name) && !slices.Contains(fallbackOption.dataSourceFallbacks, name) {
+			dataSources = append(dataSources, dataSourceFunc)
+		}
+	}
+
+	return append(dataSources, onboardedDataSources...)
+}
+
+func getResourceName(resourceFunc func() resource.Resource) string {
+	resp := resource.MetadataResponse{}
+	resourceFunc().Metadata(context.Background(), resource.MetadataRequest{ProviderTypeName: "databricks"}, &resp)
```

Comment:
We should pass the context through

Reply:
Looking more at it, will do it as a follow up to limit the size of this PR